proparse/src/main/java/org/prorefactor/proparse/antlr4/InputSource.java

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ * Copyright (c) 2003-2015 John Green
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    John Green - initial API and implementation and/or initial documentation
+ *******************************************************************************/ 
+package org.prorefactor.proparse.antlr4;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+import com.google.common.io.ByteProcessor;
+import com.google.common.io.Files;
+
+/**
+ * The bottom InputSource object for an IncludeFile is the input for the include file itself.
+ * 
+ * Each upper (potentially stacked) InputSource object is for an argument reference:
+ * <ul>
+ * <li>include file argument reference or reference to scoped or global preprocessor definition
+ * <li>the input stream pointer is a pointer to a string stream object
+ * </ul>
+ * We keep a pointer to the input stream "A" in the InputSource object so that if "A" spawns a new input stream "B", we
+ * can return to "A" when we are done with "B".
+ */
+public class InputSource {
+  // TODO Almost sure those two fields are useless
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1135)